File: ServerINes/Models/Entities/Reminder.cs
using INest.Models.Enums;

namespace INest.Models.Entities
{
    public class Reminder
    {
        public Guid Id { get; set; }
        public Guid ItemId { get; set; }
        public string Title { get; set; } = string.Empty;
        public ReminderType Type { get; set; }
        public ReminderRecurrence Recurrence { get; set; }
        public DateTime TriggerAt { get; set; }
        public bool IsCompleted { get; set; }
        public Item Item { get; set; } = null!;
        public bool IsNotificationSent { get; set; }
    }
}

File: ClientINes/src/app/core/contracts/reminder.ts
import { ReminderType } from '../enums/reminder-type.enum';
import { ReminderRecurrence } from '../enums/reminder-recurrence.enum';
import { Item } from './item';

export interface Reminder {
  id: string;
  itemId: string;
  title: string;
  type: ReminderType;
  recurrence: ReminderRecurrence;
  triggerAt: string;
  isCompleted: boolean;
  sendNotification: boolean;
  sendTelegram?: boolean;
  item?: Item;
}